src/networking.c

```diff
@@ -539,6 +539,119 @@ get_mac_address(const gchar *const ifname)
 	return macaddr;
 }
 
+/*
+ * Get and store the iptable rules set in the
+ * network namespace. These need to be re-applied
+ * within the container.
+ *
+ * \param config \ref cc_oci_config
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+get_iptable_rules(struct cc_oci_config *config)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status;
+	char *args[2] = { "iptables-save", NULL };
+
+	if (! config) {
+		return false;
+	}
+
+	ret = g_spawn_sync (NULL,
+              args,
+              NULL,
+              G_SPAWN_SEARCH_PATH | G_SPAWN_STDERR_TO_DEV_NULL,
+              NULL,
+              NULL,
+              &(config->net.iptable_rules),
+              NULL,
+              &exit_status,
+              &error);
+
+	if ( !ret || exit_status) {
+		if (! ret) {
+			g_critical("Error spawning process for iptables-save:%s",
+				 error->message);
+			g_clear_error(&error);
+		}
+
+		if (! g_spawn_check_exit_status(exit_status, &error)) {
+			g_critical("Error saving iptable rules: %s",
+				error->message);
+			g_clear_error(&error);
+		}
+		g_free_if_set(config->net.iptable_rules);
+	}
+
+	return ret;
+}
+
+/*
+ * Flush all iptable rules set in network namespace.
+ * We need to do this as rules get applied to the
+ * packets crossing the tap bridge.
+ *
+ * \param config \ref cc_oci_config
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+purge_iptable_rules(struct cc_oci_config *config)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status;
+
+	/* Set the default policies for the built-in chains to ACCEPT,
+	 * flush nat and mangle tables, flush all chains(-F) and
+	 * delete all user chains(-X).
+	 */
+	char *iptables_drop_command = "iptables -P INPUT ACCEPT "
+			"&& iptables -P FORWARD ACCEPT "
+			"&& iptables -P OUTPUT ACCEPT "
+			"&& iptables -t nat -F "
+			"&& iptables -t mangle -F "
+			"&& iptables -F "
+			"&& iptables -X";
+
+	char *args[4] = { "/bin/bash", "-c", iptables_drop_command, NULL};
```

Comment:
- Again, I'd drop the `4`.
- I don't think you should rely on `bash` (added to which it's huge and slow), so `/bin/sh` (`_PATH_BSHELL`) would be preferable.
- If we really have to call out to iptables, it would be very useful to add a couple of `g_debug()` calls (before and after) so we can see the impact in the log.
- Shouldn't we be using `ip rule ...` rather than `iptables` anyway?